src/aws/osml/model_runner/api/image_request.py

```diff
@@ -21,148 +22,242 @@
+        image_id: A combined identifier for the image, usually composed of the job ID and image URL.
+        image_url: The URL location of the image to be processed.
+        image_read_role: The IAM role used to read the image from the provided URL.
+        outputs: A list of output configurations where results should be stored.
+        model_name: The name of the model to use for image processing.
+        model_invoke_mode: The mode in which the model is invoked, such as synchronous or asynchronous.
+        tile_size: Dimensions of the tiles into which the image is split for processing.
+        tile_overlap: Overlap between tiles, defined in dimensions.
+        tile_format: The format of the tiles (e.g., NITF, GeoTIFF).
+        tile_compression: Compression type to use for the tiles (e.g., None, JPEG).
+        model_invocation_role: IAM role assumed for invoking the model.
+        feature_properties: Additional properties to include in the feature processing.
+        roi: Region of interest within the image, defined as a geometric shape.
+        post_processing: List of post-processing steps to apply to the features detected.
     """
 
+    job_id: str = ""
+    image_id: str = ""
+    image_url: str = ""
+    image_read_role: str = ""
+    outputs: List[Dict[str, Any]] = field(default_factory=list)
+    model_name: str = ""
+    model_invoke_mode: ModelInvokeMode = ModelInvokeMode.NONE
+    tile_size: ImageDimensions = DEFAULT_TILE_SIZE
+    tile_overlap: ImageDimensions = DEFAULT_TILE_OVERLAP
+    tile_format: ImageFormats = ImageFormats.NITF
+    tile_compression: ImageCompression = ImageCompression.NONE
+    model_invocation_role: str = ""
+    feature_properties: List[Dict[str, Any]] = field(default_factory=list)
+    roi: Optional[BaseGeometry] = None
+    post_processing: List[MRPostProcessing] = field(
+        default_factory=lambda: [
+            MRPostProcessing(step=MRPostprocessingStep.FEATURE_DISTILLATION, algorithm=FeatureDistillationNMS())
+        ]
+    )
+
     def __init__(self, *initial_data: Dict[str, Any], **kwargs: Any):
         """
-        This constructor allows users to create these objects using a combination of dictionaries
+        Initializes an ImageRequest instance using a combination of dictionaries
         and keyword arguments.
 
-        :param initial_data: Dict[str, Any] = dictionaries that contain attributes/values that map to this class's
-                             attributes
-        :param kwargs: Any = keyword arguments provided on the constructor to set specific attributes
+        :param initial_data: Dictionaries containing attributes and their values to set on the instance.
+        :param kwargs: Additional keyword arguments to set specific attributes on the instance.
         """
-        default_post_processing = [
+        self._initialize_attributes()
+        self._set_attributes_from_data(*initial_data, **kwargs)
+
+    def _initialize_attributes(self):
```

Comment:
Why do we need to set the initial values when they are already set when the class is instantiated?  If we changed a default value we would have to do it in two places to avoid confusion.

Reply:
Good catch - will remove and test.